### PR TITLE
Add Carthage Support

### LINKED ISSUE #17: Carthage support?
I don't like the way CocoaPods works that is to intrusive. I tried to build this library using carthage, but it doesn't build. Is it possible to make this library to support carthage? Here is the error message when I tried to build with carthage:

```
*** Building scheme "JWT" in JWT.xcworkspace
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution Check\ Pods\ Manifest.lock /Users/elgs/Library/Developer/Xcode/DerivedData/JWT-aqboxwyfwmewxqhiljchyebefxbh/Build/Intermediates/JWT.build/Release/JWT.build/Script-4B9936F743EA727EA58353BD.sh
(1 failure)
error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.
A shell task failed with exit code 65:
** BUILD FAILED **


The following build commands failed:
	PhaseScriptExecution Check\ Pods\ Manifest.lock /Users/elgs/Library/Developer/Xcode/DerivedData/JWT-aqboxwyfwmewxqhiljchyebefxbh/Build/Intermediates/JWT.build/Release/JWT.build/Script-4B9936F743EA727EA58353BD.sh
(1 failure)
```

### PR DESCRIPTION
**Pull request by tternes**
This change set provides Carthage support for the library, which would resolve #17.

I've attempted to follow the example of [AlamofireImage](https://github.com/Alamofire/AlamofireImage) for how to best satisfy dependencies while not requiring integrators to use a particular dependency management system. To accomplish this goal, I've removed internal use of Pods within JSONWebToken, and instead satisfy the CryptoSwift dependency with a submodule. I've added a Cartfile that handles the dependency chain for Carthage users, and attempted to verify CocoaPods support with a local spec. I've also updated the Travis tests and run them against my fork to make sure those continue working.

As a Carthage user, I'm most concerned about having possibly broken CocoaPods, so feedback in that area would be greatly appreciated.

As a look ahead, I'm also planning to submit changes to add support for an iOS framework target, as well as code changes to support custom header values and signing key formats. I'll also submit a follow-up PR with changes for the README.

### CODE REVIEW

**Comment by tternes** [comment]
Any chance you'll be able to review this in the next few days, @kylef? I have a couple other changes queued up that I've held until this merges.

Thanks!

File: .travis.yml [hunk 1]
```diff
@@ -5,5 +5,5 @@ before_install:
 - gem install xcpretty --no-document
 script:
 - set -o pipefail
+- git submodule update --init --recursive
 - xcodebuild -workspace JWT.xcworkspace -scheme JWT test -sdk macosx | xcpretty -c
-- pod lib lint
```
Comment (kylef) [inline .travis.yml]: Can you restore this line, it's verifying if the pod spec is valid for CocoaPods.

File: JWT.xcworkspace/contents.xcworkspacedata [hunk 2]
```diff
@@ -4,7 +4,4 @@
    <FileRef
       location = "group:JWT.xcodeproj">
    </FileRef>
-   <FileRef
-      location = "group:Pods/Pods.xcodeproj">
-   </FileRef>
 </Workspace>
```
Comment (kylef) [inline JWT.xcworkspace/contents.xcworkspacedata]: Perhaps the workspace can be removed entirely since it's no longer used?

**Comment by tternes** [comment]
@kylef I get the following error when I run `pod lib lint --no-clean --verbose` locally:

```
** BUILD SUCCEEDED **

 -> JSONWebToken (1.4.2)
    - WARN  | url: The URL (http://twitter.com/kylefuller) is not reachable.
```

I've taken care of the other comments.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+# fastlane
+#
+# It is recommended to not store the screenshots in the git repo. Instead, use fastlane to re-generate the 
+# screenshots whenever they are needed.
+# For more information about the recommended setup visit:
+# https://github.com/fastlane/fastlane/blob/master/docs/Gitignore.md
+
+fastlane/report.xml
+fastlane/screenshots
```
Comment (kylef) [inline .gitignore]: We don't use fast lane and a bunch of things in this gitignore, can you strip them out of this gitignore please?

Comment (kylef) [inline .gitignore]: We don't use fast lane and a bunch of things in this gitignore, can you strip them out of this gitignore please?

File: Cartfile [hunk 4]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (kylef) [inline Cartfile]: Can we remove this empty line? Unless is has some particular use in Carthage.

**Comment by kylef** [comment]
Okay, just two remaining comments. I've fixed the issue with the podspec in 6df50e2, so rebasing on-top of master should solve this. Would you be able to address them and then squash these commits?

I have one concern, that the Xcode project only has an OS X Target. From what I understand, that means Carthage will only be able to integrate this library into OS X and it won't work for iOS, tvOS and watchOS users. Is this a problem for you?

**Comment by tternes** [comment]
> I have one concern, that the Xcode project only has an OS X Target.

I have another PR ready for iOS / tvOS targets that should take care of that.

**Comment by tternes** [comment]
@kylef The pod lint step passes for me locally, but it looks like Travis has an issue with it. Any chance you can give me some guidance on that, since I'm not as familiar with what pods might be upset about?

Thanks so much for your help with this.

**Pull request merged**